### PR TITLE
[MB-6846] Customer Onboarding Redirection

### PR DESCRIPTION
**Pull request by suzubara**
## Description

The goal of this PR is to simplify the code that handles redirecting a customer user to the last profile wizard step they've completed.

- I added a `selectServiceMemberProfileState` selector, constant values for each state, and a util function that determines the last profile step they should be directed to (or to the Home page if they have completed all steps).
- I created a higher-order component `requireCustomerState` that uses the above code to redirect a user based on their state (currently only used on the Home & PPM landing pages) when a page mounts. This is used in place of the formerly used `getNextIncompletePage` function, and will only perform redirects to pages that the user has to complete before they can access the Home page.
  - I wrote this as a HOC so it can easily be applied to other pages, and use hooks for Redux interactions regardless of which pages are class components or functional components. It was also a lot easier to test Redux interactions using the `useDispatch` hook instead of `connect`
  - Right now it just does a simple `===` check to see if the user's state matches the state that is required by the component, but this can be expanded upon if more complex logic is needed (I'm thinking of upcoming work where we want to prevent a user from re-accessing profile wizard steps once they've completed them).

## Reviewer Notes

To test, log in as new and/or incomplete service members and verify that attempting to reach the Home page (`/`) or PPM landing page (`/ppm`) redirects you to the last step you left off at. You can create service member and click on the MilMove logo in the header at each step to verify that you are redirected back to where you left off in the flow.

You should also be able to use the back/next buttons at each part of the profile wizard without changes.

Once you've completed all steps through backup contacts, you should be able to access both the Home page (`/`) and PPM landing page (`/ppm`) without being redirected.

The last AC mentioned in the JIRA story ("Prevent SM from accessing profile steps if complete") will be completed in [this ticket](https://dp3.atlassian.net/browse/MB-6839).

## Setup

Run the app locally:
```
make server_run
make client_run
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-6846) for this change

### CODE REVIEW

**Comment by suzubara** [comment]
The Happo failures are due to the Home page component no longer importing anything from `getWorkflowRoutes` (which was inadvertently importing CSS files that contain global styles), resulting in those CSS files no longer being imported into Storybook. I am double checking with @amanda5iu but I'm pretty sure the changes shown here are what we ultimately want. Because `getWorkflowRoutes` is still used throughout the app, I believe the CSS changes are only effective in Storybook, and there is no visual diff in the app itself. This should be resolved with future CSS organizational work.

File: src/scenes/MyMove/index.jsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ export class CustomerApp extends Component {
   }
 
   componentDidMount() {
-    const { loadUser, loadInternalSchema, initOnboarding } = this.props;
+    const { loadUser, initOnboarding, loadInternalSchema } = this.props;
```
Comment (kimallen) [inline src/scenes/MyMove/index.jsx:71]: I'm curious why the switch- is it convention to put them in alphabetical order?

**Review by kimallen — APPROVED** [review]
Nice!  I like how you restructured this- it's much easier to follow than before.
I walked through all the pages and I am redirected as expected throughout the profile setup.

File: src/scenes/MyMove/index.jsx (hunk 1 above)
Comment (suzubara) [reply src/scenes/MyMove/index.jsx:71]: purely accidental, I had been trying out a few different things which involved removing this and then putting it back, lol

**Review by amanda5iu — APPROVED** [review]

**Pull request merged**